src/api/app/controllers/webui/packages/files_controller.rb

```diff
@@ -12,57 +12,44 @@ def new
       def create
         authorize @package, :update?
 
-        file = params[:file]
-        file_url = params[:file_url]
-        filename = params[:filename]
+        files = params[:files]&.map { |f| { file: f, filename: params[:filenames][f.original_filename] || f.original_filename } }
```

Comment:
:thought_balloon: Looks like this can be pushed into a `before_action`

Reply:
Also, what about [StrongParameters](https://api.rubyonrails.org/classes/ActionController/StrongParameters.html)?